lmdeploy/pytorch/configurations/default.py

```diff
@@ -23,6 +23,15 @@ def build(cls, hf_config, model_path: str = None):
         if use_sliding_window:
             sliding_window = getattr(hf_config, 'sliding_window',
                                      sliding_window) or -1
+        tp = kwargs.get('tp', 1)
+        # update num_kv_heads for tp mode
+        if tp > 1 and tp > num_key_value_heads:
+            assert tp % num_key_value_heads == 0
+            n_replicate = tp // num_key_value_heads
+            hf_config.num_replicate_key_value_heads = n_replicate
+            num_key_value_heads = tp
+            hf_config.num_key_value_heads = tp
```

Comment:
Duplicate with chatglm model. Shall we wrap it into a function since there might be more models using this?